### PR TITLE
Add Dockerfile sample

### PR DESCRIPTION
**Pull request by jhanca-robotecai**
Add Dockerfile sample in response to #19 

Tested with the following pairs: 
- humble+jammy 
- jazzy+noble

~~This PR is a draft, as it waits for the changes in the documentation. The changes in the documentation can be either added directly to this PR or as a separate PR.~~

### CODE REVIEW

**Comment by jhanca-robotecai** [comment]
Added simple README, it should be heavily updated.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,166 @@
+ARG ROS_VERSION=jazzy
+ARG UBUNTU_VERSION=noble
+
+FROM ros:${ROS_VERSION}-ros-base-${UBUNTU_VERSION}
```
Comment (pijaro) [inline docker/Dockerfile]: You can use `ros:humble-ros-base` or `ros:jazzy-ros-base` tags and eliminate the Ubuntu version argument.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+ARG PROJECT_DEMO_REPO=https://github.com/RobotecAI/rai-rosbot-xl-demo.git
+ARG PROJECT_DEMO_BRANCH=main
+ARG PROJECT_DEMO_COMMIT=HEAD
+ENV PROJECT_DEMO_ROOT=$WORKSPACE/RAIROSBotXL
+ENV PROJECT_DEMO_PROJECT=$PROJECT_DEMO_ROOT/Project
+
+ENV LANG=en_US.UTF-8
+
+# Setup time zone and locale data (necessary for SSL and HTTPS packages)
+RUN apt-get update \
```
Comment (pijaro) [inline docker/Dockerfile]: This is already a part of ROS core docker image.

File: README.md [hunk 3]
```diff
@@ -27,11 +25,46 @@ Simulation environment using O3DE.
...
+The _Dockerfile_ defined in [docker folder](./docker) will prepare the appropriate ROS 2 distribution based environment and build the components necessary to run the demo project simulator through the O3DE engine.
+
+#### Requirements
+
+* [Hardware requirements of o3de](https://www.o3de.org/docs/welcome-guide/requirements/)
+* Ubuntu 22.04 (Jammy)
+* At least 60 GB of free disk space
+* Docker installed and configured
+  * **Note** It is recommended to have Docker installed correctly and in a secure manner so that the docker commands in this guide do not require elevated privileges (sudo) in order to run them. See [Docker Engine post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) for more details.
+* [NVidia container toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker)
```
Comment (pijaro) [inline README.md]: Nitpick: `NVIDIA`, not `NVidia` :)

Comment (jhanca-robotecai) [reply README.md]: image

File: README.md [hunk 4]
```diff
@@ -27,11 +25,46 @@ Simulation environment using O3DE.
...
+* Docker installed and configured
+  * **Note** It is recommended to have Docker installed correctly and in a secure manner so that the docker commands in this guide do not require elevated privileges (sudo) in order to run them. See [Docker Engine post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) for more details.
+* [Nvidia container toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker)
+
+#### Building the Full Docker Image
+
+The `Dockerfile` supports defining which version of Ubuntu+ROS to base the docker container on, and by default will support Ubuntu 24.04 (noble) with the ROS 2 Jazzy distribution. It will build a Docker image that will contain the Editor and the demo launcher. Use the following command to build the image:
+
+```
+docker build -t rai-rosbot -f Dockerfile .
```
Comment (ycheng517) [inline README.md:50]: Dockerfile is located in the `docker` directory

**Review by ycheng517 — APPROVED** [review]
Thanks, I can run both the release binary and the simulation scripts using the docker image!

File: README.md (hunk 4 above)
Comment (jhanca-robotecai) [reply README.md:50]: Added, thank you.

**Review by pijaro — APPROVED** [review]

**Pull request merged**